contracts/modules/Migrator.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {MigratorImmutables} from '../modules/MigratorImmutables.sol';
+import {INonfungiblePositionManager} from '@uniswap/v3-periphery/contracts/interfaces/INonfungiblePositionManager.sol';
+import {TokenAuthorized} from '../libraries/TokenAuthorized.sol';
+
+abstract contract Migrator is MigratorImmutables {
+    using TokenAuthorized for address;
```

Comment:
Thinking about naming for this, i think i'd like if it included the word `cache`
like
`if (spender.isAuthorizationCached(tokenId))` and `if (authorized.cacheAuthorization(tokenId))`
i feel like those read more easily than
`is (spender.getAuthorized(tokenId))` and `if (authorized) spender.setAuthorized(tokenId)`

Then library called... `TokenAuthorizationCache`? or something?